source/jormungandr/jormungandr/interfaces/v1/Journeys.py

```diff
@@ -217,6 +224,68 @@ def wrapper(*args, **kwargs):
         return wrapper
 
 
+# Add TAD deep links in each section of type on_demand_transport if network contains app_code in codes
+class add_tad_links(object):
+    def __call__(self, f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            objects = f(*args, **kwargs)
+            if objects[1] != 200:
+                return objects
+            if "journeys" not in objects[0]:
+                return objects
+
+            for j in objects[0]['journeys']:
+                if "sections" not in j:
+                    continue
+                for s in j['sections']:
+                    # For a section with type = on_demand_transport
+                    if s.get('type') == 'on_demand_transport':
+                        # get network uri from the link
+                        networks = [link['id'] for link in s.get('links', []) if link['type'] == "network"]
+                        if len(networks) == 0:
+                            continue
+                        network_id = networks[0]
+                        region = kwargs.get('region')
+                        if region is None:
+                            continue
+
+                        # Get the Network details and verify if it contains codes with type = "app_code"
+                        instance = i_manager.instances.get(region)
+                        network_details = instance.ptref.get_objs(
+                            type_pb2.NETWORK, 'network.uri={}'.format(network_id)
+                        )
+                        network_dict = protobuf_to_dict(next(network_details))
```

Comment:
I don't know if necessary to do "protobuf_to_dict"

Reply:
We need to transform as they don't have same structures